juniper/src/vespa/juniper/sumdesc.cpp

```diff
@@ -40,6 +68,41 @@ bool nonwordchar(const unsigned char* s)
     }
 }
 
+bool
+il_ann_anchor_char(const unsigned char* s)
+{
+    unsigned char c = *s;
```

Comment:
Please move the content of this function into a helper function that both il_ann_anchor_char(), il_ann_separator_char() and il_ann_terminator_char() can call.